site/specs/webrtc.yml

```diff
@@ -2421,6 +2520,201 @@ components:
+    sipUpdateDialogRequestBody:
+      type: object
+      description: Update a SIP Dialog owned by a Participant.
+      properties:
+        state:
+          type: string
+          description: the name of the state that the Dialog has entered
+          example: ANSWERED
+        tag:
+          type: string
+          description: >-
+            User defined tag to associate with the dialog.  Must not contain
+            PII.
+          example: sip-patron-dialog
+    eventBody:
+      oneOf:
+        - $ref: '#/components/schemas/sipDialogEventBody'
+        - $ref: '#/components/schemas/onLeaveEventBody'
+        - $ref: '#/components/schemas/onDisconnectEventBody'
+        - $ref: '#/components/schemas/onConnectEventBody'
+      discriminator:
+        propertyName: event
+        mapping:
+          onSipDialogStateChange: '#/components/schemas/sipDialogEventBody'
+          onLeave: '#/components/schemas/onLeaveEventBody'
+          onDisconnect: '#/components/schemas/onDisconnectEventBody'
+          onConnect: '#/components/schemas/onConnectEventBody'
+    baseEventBody:
+      type: object
+      description: the common properties shared by all events
+      required:
+        - event
+      properties:
+        event:
+          type: string
+          description: >-
+            the name of the event, and the condition that describes the body
+            that will be present
+          example: onSipDialogStateChange
+          enum:
+            - onSipDialogStateChange
+            - onLeave
+            - onDisconnect
+            - onConnect
+        timestamp:
+          type: string
+          description: the epoch UTC time, in milliseconds, when the event happened
+          example: '1628786234676'
+        participantId:
+          type: string
+          description: the ID of Participant on which the event was detected
+          example: be199214-1a03-402a-a7b6-d002470a0465
+        dialogId:
+          type: string
+          description: the ID of Participant on which the event was detected
+          example: be199214-1a03-402a-a7b6-d002470a0465
+        tag:
+          type: string
+          description: >-
+            the tag associated with the Participant on which the event was
+            detected.
+          example: hello-world-browser
+      discriminator:
+        propertyName: event
+    dialogEventData:
+      type: object
+      description: >
+        The dialogEventData element contains information about the event that
+        triggered the transition to the  current Dialog state. This data is
+        dependent on extrinsic events that impact the Dialog. An understanding
+        of the details of the Dialog Event may affect the business logic choices
+        for the receiving application.
+      properties:
+        eventReason:
+          type: string
+          description: A summary of why the event happened
+          example: FAR END BUSY
+        eventErrorReason:
```

Comment:
Could there not be a reason/description for a non error event? How about just calling this description?

Reply:
Let's discuss this entire payload when we review with the team and DX.
The error code is something that could be added later for example.  I'm not sure that the reason should need an explanation in every event, so that one needs some discussion.